src/content/documentation/develop/differences-between-desktop-and-android-extensions.md

```diff
@@ -72,49 +66,58 @@ A detailed list of the WebExtension APIs supported in Firefox for Android is pro
-
-### Effect on your add-on UI
-
-These differences impact the way you expose your add-on in the Firefox UI. The most common option, adding a button for your add-on to the Firefox toolbar with [`browserAction`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/browserAction), is not available (at least until Firefox 55). Nor can you expose your add-on through a sidebar or context menu. You will, therefore, use an address bar button (through the `manifest.json` [`page_action`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/page_action) key and [`pageAction`](https://developer.mozilla.org//Add-ons/WebExtensions/API/pageAction) API) remembering that by default this button is hidden and must be shown programmatically.
-
-The features of [`pageAction`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction) are also reduced in Firefox for Android. The `manifest.json` key [page_action](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/page_action) enables you to define the button icon and a popup. You then have use of [`pageAction.show()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction/show) and [`pageAction.hide()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction/hide) however, once ‘shown’, note that the address bar button is visible in all tabs (unlike the desktop behavior, where the button is shown only for a specified tab.) But you will still be able to hide the pageAction using [`pageAction.hide()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction/hide) on a specific tab (say, for example, you wish to hide your extension's page action icon in `about:addons` or `about:memory` tabs) And you can set a listener to [`pageAction.onClicked()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction/onClicked). [`pageAction.setPopup()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction/setPopup) and [`pageAction.getPopup()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction/getPopup) are also available, so you can update the popup or create a popup once the add-on is running.
+- [`sidebarAction`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/sidebarAction) and the related [`sidebar_action`](https://developer.mozilla.org/Add-ons/WebExtensions/manifest.json/sidebar_action) `manifest.json` key, due to the limited screen real estate on Android devices sidebars, such as the browser history, are presented in full browser tabs. Where possible, you should move any sidebar content to tabs.
+- [`windows`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/windows), as there is only one Firefox on Android 'window', and the browser cannot open or otherwise manipulate additional browser windows.
```

Comment:
@rpl it's my understanding that this is not entirely correct. I thought we had (at least) 2 windows for regular browsing mode and private browsing mode? It's correct that we don't expose the `windows` API, though.

Reply:
@willdurand on mobile, isn't it a case of regular browsing, and private tabs rather than windows?